src/joystream-lib/api.ts

```diff
@@ -424,52 +427,130 @@ export class JoystreamJs {
+      cb
+    )
   }
 
   async updateChannel(
     channelId: ChannelId,
     memberId: MemberId,
     inputMetadata: CreateChannelMetadata,
-    inputAssets: ChannelAssets,
+    inputAssets: InputAssets,
     cb?: ExtrinsicStatusCallbackFn
   ): Promise<ExtrinsicResult<ChannelId>> {
+    const { avatarContent, coverContent } = this._prepareChannelAssets(inputAssets)
+    const assets: { [field: string]: [NewAsset, string] | undefined } = {
+      avatarAsset: avatarContent || undefined,
+      coverAsset: coverContent || undefined,
+    }
+    const contentIds = {
+      avatarId: assets.avatarAsset ? assets.avatarAsset[1] : '',
+      coverId: assets.coverAsset ? assets.coverAsset[1] : '',
+    }
     await this.ensureApi()
 
-    return this._createOrUpdateChannel(channelId, memberId, inputMetadata, inputAssets, cb)
+    return this._createOrUpdateChannel(
+      channelId,
+      memberId,
+      inputMetadata,
+      { avatar: assets.avatarAsset && assets.avatarAsset[0], cover: assets.coverAssets && assets.coverAssets[0] },
+      contentIds,
+      cb
+    )
+  }
+
+  private _prepareVideoAssets(inputAssets: InputAssets) {
+    return {
+      videoContent: inputAssets.video && this._createFileAsset(inputAssets.video),
+      thumbnailContent: inputAssets.thumbnail && this._createFileAsset(inputAssets.thumbnail),
+    }
   }
 
   async createVideo(
     memberId: MemberId,
     channelId: ChannelId,
     inputMetadata: CreateVideoMetadata,
-    inputAssets: VideoAssets,
+    inputAssets: InputAssets,
     cb?: ExtrinsicStatusCallbackFn
   ): Promise<ExtrinsicResult<VideoId>> {
+    const {
+      videoContent: [video, videoId],
+      thumbnailContent: [thumbnail, thumbnailId],
+    } = this._prepareVideoAssets(inputAssets)
     await this.ensureApi()
 
-    return this._createOrUpdateVideo(null, memberId, channelId, inputMetadata, inputAssets, cb)
+    return this._createOrUpdateVideo(
+      null,
+      memberId,
+      channelId,
+      inputMetadata,
+      { video, thumbnail },
+      { videoId, thumbnailId },
+      cb
+    )
   }
 
   async updateVideo(
-    videoId: VideoId,
+    updatedVideoId: VideoId,
     memberId: MemberId,
     channelId: ChannelId,
     inputMetadata: CreateVideoMetadata,
-    inputAssets: VideoAssets,
+    inputAssets: InputAssets,
     cb?: ExtrinsicStatusCallbackFn
   ): Promise<ExtrinsicResult<VideoId>> {
+    const { videoContent, thumbnailContent } = this._prepareVideoAssets(inputAssets)
+    const assets: { [field: string]: [NewAsset, string] | undefined } = {
+      videoAsset: videoContent || undefined,
+      thumbnailAsset: thumbnailContent || undefined,
+    }
+    const contentIds = {
+      videoId: assets.videoAsset ? assets.videoAsset[1] : '',
+      thumbnailId: assets.thumbnailAsset ? assets.thumbnailAsset[1] : '',
+    }
     await this.ensureApi()
 
-    return this._createOrUpdateVideo(videoId, memberId, channelId, inputMetadata, inputAssets, cb)
+    return this._createOrUpdateVideo(
```

Comment:
There isn't a good place for this comment so just leaving it here - the error with cancelled signature is still there